src/test/java/spoon/test/support/ResourceTest.java

```diff
@@ -90,4 +90,31 @@ public void testFilteringFolder() throws Exception {
 		assertEquals(nbAll - nbPackageProtected, mb3.getFactory().getModel().getAllTypes().size());
 	}
 
+	@Test
+	public void testPerformanceOfVirtualFolder() throws Exception {
+		String dir = "src";
+		FileSystemFolder fileSystemFolder = new FileSystemFolder(new File(dir));
+
+		SpoonFolder folder = new VirtualFolder();
+		folder.addFolder(fileSystemFolder);
+
+		//Measure how long it needs to return all files
+		long startTime = System.currentTimeMillis();
+		folder.getAllFiles();
+		assertTrue("The VirtualFolder.getAllFiles() is too slow", (System.currentTimeMillis() - startTime)<1000);
```

Comment:
Do you check the times needed by execution of each test? I do not know your testing environment, but some tools reports a warning/error even when test passes, but needs much longer time then previous run of the same test. If you have such execution time comparator, then it would make sense to keep the test there - I would just remove assert which checks absolute time.